### PR TITLE
Add openapi spec for delete backup api

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

Define spec for delete backup api. The spec aligns with the api discussed [here](https://docs.google.com/document/d/1pITW-pvUiGuJ8kIeR1vqjnoqmhs0unnNN955jUkhF6w/edit#heading=h.8n6adu75g6g3)

## Related issues

related #10209

### CODE REVIEW

File: dist/src/main/resources/api/backup-management-api.yaml [hunk 1]
```diff
@@ -88,8 +82,31 @@ paths:
...
+        - $ref: '#/components/parameters/BackupId'
+      responses:
+        '204':
+          description: Backup is deleted
+        '500':
+          description: Internal error
+          $ref: '#/components/responses/Error'
+        '502':
+          description: Operation failed because connection to the broker or to the backup storage is lost.
+          $ref: '#/components/responses/Error'
```
Comment (Zelldon) [inline dist/src/main/resources/api/backup-management-api.yaml:98]: :thinking: I think what you want here is then `504`

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses 

> [504 Gateway Timeout](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/504)
> This error response is given when the server is acting as a gateway and cannot get a response in time.
> 

Whereas 502 is:


> [502 Bad Gateway](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/502)
> This error response means that the server, while working as a gateway to get a response needed to handle the request, got an invalid response.
>

File: dist/src/main/resources/api/backup-management-api.yaml [hunk 2]
```diff
@@ -88,8 +82,31 @@ paths:
         '502':
           description: Operation failed because connection to the broker or to the backup storage is lost.
           $ref: '#/components/responses/Error'
+    delete:
+      summary: Delete a backup
+      description: Delete a backup with the given id
+      parameters:
+        - $ref: '#/components/parameters/BackupId'
```
Comment (Zelldon) [inline dist/src/main/resources/api/backup-management-api.yaml:89]: Interesting thing. I never have seen such a thing :D

**Review by ChrisKujawa — APPROVED** [review]
I guess :+1: :D 

I had some comment but I guess I'm too late to the party. :)

File: dist/src/main/resources/api/backup-management-api.yaml (hunk 1 above)
Comment (deepthidevaki) [reply dist/src/main/resources/api/backup-management-api.yaml:98]: Oh.. yes. `504` suits better in this case. Thanks for pointing this out. I will check with other teams.

Comment (deepthidevaki) [reply dist/src/main/resources/api/backup-management-api.yaml:98]: Will update it after syncing with other teams. For now, I will merge this.

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**